JujuWindowsUtils/JujuWindowsUtils.psm1

```
@@ -352,18 +352,38 @@ function Install-WindowsFeatures {
     )
     PROCESS {
         $rebootNeeded = $false
-        foreach ($feature in $Features) {
-            $state = Start-ExecuteWithRetry -Command {
-                Install-WindowsFeature -Name $feature -ErrorAction Stop
-            }
-            if ($state.Success -eq $true) {
-                if ($state.RestartNeeded -eq 'Yes') {
-                    $rebootNeeded = $true
+            foreach ($feature in $Features) {
+                if (Get-IsNanoServer) {
+                    try {
+                        $featureStat = Start-ExecuteWithRetry -Command {
+                            Get-WindowsOptionalFeature -Online -FeatureName $feature
+                        }
+                    } catch {
```

No need to wrap the original exception. You can let it throw and bubble up.